src/utils/mod.rs

```diff
@@ -4,3 +4,7 @@ pub fn first_or<T>(a: Option<T>, b: Option<T>) -> Option<T> {
         None => b,
     }
 }
+
+pub fn invert<T, E>(x: Option<Result<T, E>>) -> Result<Option<T>, E> {
+    x.map_or(Ok(None), |v| v.map(Some))
+}
```

Comment:
Function naming suggestion:
```suggestion
pub fn opt_result_of_result_opt<T, E>(x: Option<Result<T, E>>) -> Result<Option<T>, E> {
    x.map_or(Ok(None), |v| v.map(Some))
}
```

Reply:
Done